emily/handler/src/database/entries/deposit.rs

```diff
@@ -411,16 +418,48 @@ impl TryFrom<UpdateDepositsRequestBody> for ValidatedUpdateDepositsRequest {
     type Error = Error;
     fn try_from(update_request: UpdateDepositsRequestBody) -> Result<Self, Self::Error> {
         // Validate all the depoit updates.
-        let deposits = update_request
+        let mut deposits: Vec<_> = update_request
             .deposits
             .into_iter()
             .map(|i| i.try_into())
             .collect::<Result<_, Error>>()?;
+
+        // Order the updates by order of when they occur so that it's
+        // as though we got them in chronological order.
+        deposits.sort_by_key(|update: &ValidatedDepositUpdate| update.event.stacks_block_height);
```

Comment:
Will this mess up the order of the returned deposits in `UpdateDepositsResponse`? wrt the order in the client request.

Reply:
I actually assumed it wouldn't matter, and that it wouldn't need to be in the same order, but if we care about that I can add that change

Reply:
Fixed ~